### ISSUE TITLE
Sylius crashes when sending preflight request (OPTIONS) in dev & debug mode

### ISSUE THREAD

**Issue by PILLOWPET** [Sylius/Sylius#14829]
**Sylius version affected**: 1.12.4

**Description**  
**Main Problem:** Sylius crashes when sending preflight request (OPTIONS) in dev & debug mode

We are working on a Next/Sylius application and I am having trouble using both the frontend & backend locally with debug & dev mode.

1. Why this happens?

Whenever I try to send a request from the frontend to the backend, the Preflight request fails (the OPTIONS request sent before the POST for instance).
=> Because an error is thrown `There is currently no session available` (`SessionNotFoundException`) which is normal for an `OPTIONS` request I guess
=> Because Sylius tries to check via the `SessionProvider::getSession` if the user has a cart for the specific channel (`CartSessionStorage::hasForChannel`)
=> Because the `SessionAndChannelBasedCartContext` is part of the `CompositeCartContext` and is called in the `CartCollector` class
=> Because I am assuming Sylius adds a `CartCollector` to the collectors of the Symfony Profiler to display debug data regarding the `cart` 

2. Why this does not happen in production?

In production we have the profiler disabled with `APP_DEBUG` set to 0 and `APP_ENV` set to `prod`.

**Steps to reproduce**  
1. Setup a frontend on `localhost:3000` for example
2. Send a request to your backend on `localhost:8000`
3. Send an OPTIONS request on any `v2/shop/` route with `APP_DEBUG` set to 1, `APP_ENV` set to `dev` and the profiler in `dev/framework.yaml` set to
```
framework:
    profiler: { only_exceptions: false }
```

**Possible Solution**  
Right now we are removing the `framework.yaml` when we want to work with the API (headless mode) and we add it back when working in the admin

I'd be glad to know if someone encountered this using a similar setup or if it is something that has never been spotted with Sylius.

**Comment by LucasLaurensMLP** [comment]
Hi @PILLOWPET 
 
I have indeed the same error
I tried on another `Symfony` project with `ApiPlatform` and it works fine

But not on my Sylius API

I'm still looking for

For information the problem does not occur on the GET but on the POST

**Comment by olimination** [comment]
I've got the same issue with Sylius v1.12.5, Symfony 6 and PHP 8.2.

For the moment I just have disabled the profiler via the `framework.yaml`:
```
when@dev:
    framework:
        profiler: { enabled: false, only_exceptions: false }
```

**Comment by LucasLaurensMLP** [comment]
I was able to move forward a bit and partially unblock myself.
The `NelmioCorsBundle` package works however Sylius API is waiting for a session for it to fully work. So by creating an empty session from the `index.php` I have no more errors. However I don't find this solution very clean but rather intermediate, so I keep looking.

**Comment by LucasLaurensMLP** [comment]
> I've got the same issue with Sylius v1.12.5, Symfony 6 and PHP 8.2.
> 
> For the moment I just have disabled the profiler via the `framework.yaml`:
> 
> ```
> when@dev:
>     framework:
>         profiler: { enabled: false, only_exceptions: false }
> ```

Currently it doesn't work for me i have to add an empty session

**Comment by LucasLaurensMLP** [comment]
I am on a NextJs project. Another solution I could find is to create a proxy on the front side to rewrite the url

**Comment by LucasLaurensMLP** [comment]
So the final solution I could apply is to follow this [issue](https://github.com/Sylius/Sylius/issues/11557). Then I changed the priorities of `sylius.context.cart.new` and `sylius.context.cart.new_shop_based` to be lower than `sylius.context.cart.session_and_channel_based` which I had already changed. Also I managed my `NelmioCorsBundle` parameters and I have a `200` on the `OPTIONS` request before the `POST` one. No need to proxy or touch the `index.php` file.
